backend/videoService/opentokVideoService.ts

```diff
@@ -36,49 +43,55 @@ class OpenTokVideoService implements VideoService {
     });
   }
 
-  generateToken(sessionId: string): { token: string; apiKey: string } {
+  generateToken(sessionId: string, tokenRole: TokenRole): { token: string; apiKey: string } {
     const token = this.opentok.generateToken(sessionId, {
-      role: OpenTokVideoService.getTokenRole(),
+      role: OpenTokVideoService.getTokenRole(tokenRole),
     });
     return { token, apiKey: this.config.apiKey };
   }
 
-  startArchive(roomName: string, sessionId: string): Promise<Archive> {
-    return new Promise((resolve, reject) => {
-      this.opentok.startArchive(
-        sessionId,
-        {
-          name: roomName,
-          resolution: '1920x1080',
-          layout: {
-            // In multiparty archives, we use the 'bestFit' layout to scale based on the number of streams. For screen-sharing archives,
-            // we select 'horizontalPresentation' so the screenshare stream is displayed prominently along with other streams.
-            // See: https://developer.vonage.com/en/video/guides/archive-broadcast-layout#layout-types-for-screen-sharing
-            type: 'bestFit',
-            screenshareType: 'horizontalPresentation',
+  async startArchive(roomName: string, sessionId: string, tokenRole: TokenRole): Promise<Archive> {
```

Comment:
Note: the `startArchive` and `stopArchive` were both marked with `async` to match the `vonageVideoService`. They do resolve as a promise after all